src/common/regions/region_db.ts

```diff
@@ -0,0 +1,66 @@
+import { sortBy, takeRight } from 'lodash';
+
+import { Region, County, State, FipsCode } from 'common/regions';
+
+class RegionDB {
+  private regions: Region[];
+
+  constructor(public states: State[], public counties: County[]) {
+    this.regions = [...states, ...counties];
+  }
+
+  findByFipsCode(fipsCode: FipsCode): Region | null {
+    const region = this.regions.find(region => region.fipsCode === fipsCode);
+    return region || null;
+  }
```

Comment:
I think I did this in one of my throwaway branches, but maybe we should import `statesByFips` and `countiesByFips` and add them as private properties for fast search by FIPS?

```ts
private regionsByFips: {[fipsCode:string]: Region};

// ...
this.regionsByFips = {...statesByFips, ...countiesByFips };
```
Also, it would be nice to have properties to access states and counties directly, without having to filter by type (since we already have them in the other file)

```
private states: State[];
private counties: Counties[];

// ...
this.states = states;
this.counties = counties;
```

Reply:
ooo maybe i'll take your function that you had.  if it's alright, I think we can handle that in another PR - it'll be easy to swap that in or out at any point!